### PR TITLE
[CDS-104797] GitOps release notes (V1.21)

### PR DESCRIPTION
**Pull request by vishal-av**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

**Review by SushrutHarness — DISMISSED** [review]
LGTM

File: release-notes/continuous-delivery.md [hunk 1]
```diff
@@ -45,6 +45,29 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
 
 </details>
 
+## December
+
+#### GitOps Version 1.21.0, GitOps Agent Version 0.82.0
+
+#### New Features and enhancements
+
+- Harness now supports the `prefixed_identifier` field in the GitOps Agent Terraform resource. This field provides a scope-prefixed agent identifier, such as `account.agentId` for account-level, `org.agentId` for organization-level, and `agentId` for project-level agents, enabling easier referencing in other Terraform resources. (CDS-101503)
```
Comment (mteodor) [inline release-notes/continuous-delivery.md:54]: `account.agentId` - it is not clear which part is constant and which changes, we need to indicate that `account`, `org` are constant and `agentId` changes as per agent.

File: release-notes/continuous-delivery.md [hunk 2]
```diff
@@ -45,6 +45,29 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+
+- The GitOps Applications landing page is now automatically refreshed, allowing newly created applications to appear in real-time without requiring manual refresh. (CDS-68662)
+
+- We have introduced the `allowEmptyCommit` field in the **GitOps Update Release Repo step**, enabling users to push an empty commit if desired. (CDS-103191)
```
Comment (mteodor) [inline release-notes/continuous-delivery.md:58]: can we add a link to docs for the step

**Review by krishi0408 — APPROVED** [review]

File: release-notes/continuous-delivery.md [hunk 3]
```diff
@@ -45,6 +45,30 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+
+- We have introduced the `allowEmptyCommit` field in the **GitOps Update Release Repo step**, enabling users to push an empty commit if desired. For more information, go to Harness [Update Release Repo step](/docs/continuous-delivery/gitops/pr-pipelines/gitops-pipeline-steps/#update-release-repo-step). (CDS-103191)
+
+- Addressed and resolved all known critical vulnerabilities related to the GitOps Agent and GitOps Service. (CDS-102606)
+
+- Harness has upgraded Argo from version 2.10 to 2.12.
+
+#### Fixed Issues
+
+- Previously: The agent_token field in the harness_platform_gitops_agent data source returned null, making it unusable for authentication validation. The issue is fixed. The agent_token field now correctly returns its value, and a new `isAuthenticated` field has been added to indicate the authentication status.(CDS-104468)
```
Comment (SudarshanPurohit) [inline release-notes/continuous-delivery.md:67]: Probably good to say we're talking about the Harness Terraform Provider here.

File: release-notes/continuous-delivery.md [hunk 4]
```diff
@@ -45,6 +45,30 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+#### GitOps Version 1.21.0, GitOps Agent Version 0.82.0
+
+#### New Features and enhancements
+
+- Harness now supports the `prefixed_identifier` field in the GitOps Agent Terraform resource. This field provides a scope-prefixed agent identifier, where the scope (account, org, or project) is constant, and the agentId varies based on the specific agent. 
+For example, `account.agentId` for Account-level agents, `org.agentId` for Organisation-level agents, `agentId` (without a prefix) for Project-level agents. (CDS-101503)
+
+- The GitOps Applications landing page is now automatically refreshed, allowing newly created applications to appear in real-time without requiring manual refresh. (CDS-68662)
+
+- We have introduced the `allowEmptyCommit` field in the **GitOps Update Release Repo step**, enabling users to push an empty commit if desired. For more information, go to Harness [Update Release Repo step](/docs/continuous-delivery/gitops/pr-pipelines/gitops-pipeline-steps/#update-release-repo-step). (CDS-103191)
```
Comment (SudarshanPurohit) [inline release-notes/continuous-delivery.md:59]: Previously, the step failed if there were no changes in the commit. This is helpful in setting up automated periodic PR pipelines.

Comment (mani5h-harness) [reply release-notes/continuous-delivery.md:59]: this should be in ng-manager
Are we listing changes in ng-manager here?

Comment (vishal-av) [reply release-notes/continuous-delivery.md:59]: I will add this to the ng manager release notes. was this a part of 1.68 release? @ashinsabu3 , can you confirm

File: release-notes/continuous-delivery.md [hunk 5]
```diff
@@ -45,6 +45,29 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+
+#### New Features and enhancements
+
+- Harness now supports the `prefixed_identifier` field in the GitOps Agent Terraform resource. This field provides a scope-prefixed agent identifier, such as `account.agentId` for account-level, `org.agentId` for organization-level, and `agentId` for project-level agents, enabling easier referencing in other Terraform resources. (CDS-101503)
+
+- The GitOps Applications landing page is now automatically refreshed, allowing newly created applications to appear in real-time without requiring manual refresh. (CDS-68662)
+
+- We have introduced the `allowEmptyCommit` field in the **GitOps Update Release Repo step**, enabling users to push an empty commit if desired. (CDS-103191)
+
+- Addressed and resolved all known critical vulnerabilities related to the GitOps Agent and GitOps Service. (CDS-102606)
```
[the remark this answers is not in the document]
Comment (acasian-harness) [reply release-notes/continuous-delivery.md:61]: also the version of argo

**Pull request closed without merging**